### PR TITLE
NotificationEntry: Copy line length and truncate rules

### PR DESCRIPTION
**Pull request by danrabbit**
Copies rules from https://github.com/elementary/notifications/blob/master/src/Bubble.vala#L102
* Consistent width: always 33ch wide
* Titles ellipsize, not wrap
* Body can be a maximum of 2 lines
* Forcefully remove extra new lines in body if necessary and truncate each line independently 

Also make timestamps dim label as we do in Mail

### CODE REVIEW

File: src/Widgets/NotificationEntry.vala [hunk 1]
```diff
@@ -40,33 +40,48 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
...
             body_label.wrap_mode = Pango.WrapMode.WORD_CHAR;
-            grid.attach (body_label, 0, 1, 2, 1);
+            body_label.wrap = true;
+            body_label.xalign = 0;
+
+            if ("\n" in body) {
+                string[] lines = body.split ("\n");
+                string stripped_body = lines[0] + "\n";
+                for (int i = 1; i < lines.length; i++) {
+                    stripped_body += lines[i].strip () + "";
```
Comment (jeremypw) [inline src/Widgets/NotificationEntry.vala:77]: Shouldn't the concatenated lines be separated by space rather than an empty string?

Comment (danrabbit) [reply src/Widgets/NotificationEntry.vala]: Oops, fixed. Good catch!

File: src/Widgets/NotificationEntry.vala [hunk 2]
```diff
@@ -40,33 +40,48 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
...
+
+            if ("\n" in body) {
+                string[] lines = body.split ("\n");
+                string stripped_body = lines[0] + "\n";
+                for (int i = 1; i < lines.length; i++) {
+                    stripped_body += lines[i].strip () + " ";
+                }
+
+                body_label.label = stripped_body.strip ();
+                body_label.lines = 1;
```
Comment (jeremypw) [inline src/Widgets/NotificationEntry.vala:81]: I am not sure why the number of lines is changed to 1 here when it was set to 2 earlier.

Comment (jeremypw) [reply src/Widgets/NotificationEntry.vala:81]: Its OK I get it now.

**Pull request merged**